crates/deltalake-core/src/operations/vacuum.rs

```diff
@@ -217,6 +263,16 @@ impl VacuumBuilder {
             specified_retention_millis: Some(retention_period.num_milliseconds()),
         })
     }
+
+    /// Whether a file is contained within the set of managed files.
+    fn is_file_managed(&self, managed_files: &HashSet<String>, file: &Path) -> bool {
+        return if is_absolute_path(file.as_ref()) {
+            managed_files.contains(file.as_ref())
+        } else {
+            let path = format!("{}{}", self.log_store.root_uri(), file.as_ref());
+            managed_files.contains(&path)
+        };
+    }
```

Comment:
Can the `location` of an object in the log_store ever be absolute? Or is it always relative to the table root? If so, then we wouldn't need the `is_absolute_path` check

Reply:
@Jan-Schweizer the location can be an absolute URL, and that's the case with shallow cloned tables. The [add action](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-file-and-remove-file) describes the `path` key. 

So for better or worse, we need to check for an absolute path  :smile: